compaction/compaction_task.hh

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include "tasks/task_manager.hh"
+
+class compaction_module : public tasks::task_manager::module {
+public:
```

Comment:
perhaps rename the header to task_manager_module.hh, it's living in compaction subdir anyway. 

Also worth placing compaction_module in compaction namespace, and renaming it to task_manager_module.

so compaction_manager can refer to it as compaction::task_manager_module which is more descriptive.

the names compaction_task and compaction_module overlap with what's already present in the compaction layer, which can cause confusion down the road.

Reply:
Sounds reasonable, I have some concerns about naming compatibility with other task manager's modules, though.

If we have a compaction/task_manager_module.hh, then repair/repair_task.hh should be renamed as well not to cause the confusion. 

The problem arises for node operations (which are planned to be integrated after compaction). There is no node_ops/ directory and node ops are strictly connected to repair (which already has its module). It's not clear what to do with them. We could keep them both in the same file (disambiguated thanks to namespace) or create separate directory...

What I mean is that somewhere in the future (even if not for node_ops) we may need to create an additional directory just to stick with module_name/task_manager_module.hh convention. Is it acceptable and wouldn't it create a mess?